velox/functions/lib/Re2Functions.cpp

```diff
@@ -978,7 +988,24 @@ std::pair<PatternKind, vector_size_t> determinePatternKind(StringView pattern) {
       // Ensures that pattern has a single contiguous stream of wildcard
       // characters.
       if (wildcardStart != -1) {
-        return std::make_pair(PatternKind::kGeneric, 0);
+        if (singleCharacterWildcardCount) {
+          return PatternMetadata{PatternKind::kGeneric, 0};
+        }
+        // Check if the pattern is of type kFixedCenter, so the use of regular
+        // expressions can be avoided to match the fixed string.
+        vector_size_t endIdx = i;
+        while ((i < patternLength) && (patternStr[i] == '%')) {
+          i++;
+        }
+        if (i == patternLength) {
+          return PatternMetadata{
```

Comment:
i wonder if to simplify we can move this out like others and instead introduce 
The only early exist in the loop will remain that if we have more than one fixedPattern which would remain. 

so if after the loop we have .
if (fixedPatternStart>wildcardStart &&  singleCharacterWildcardCount==0){
          return PatternMetadata{
              PatternKind::kFixedCenter,
              0,
              std::string(
                  pattern.data() + fixedPatternStart,
                  endIdx - fixedPatternStart)};
}

Reply:
Updated, thanks.